src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/ModelBinderDescriptorExtensions.cs

```diff
@@ -80,5 +80,24 @@ public static class ModelBinderDescriptorExtensions
             descriptors.Insert(index, descriptor);
             return descriptor;
         }
+
+
+        /// <summary>
+        /// Removes all instances of <typeparamref name="TInstance"/> from a list of ModelBinderDescriptors.
+        /// </summary>
+        /// <typeparam name="TInstance">A type that represents an <see cref="IModelBinder"/>.</typeparam>
+        /// <param name="descriptors">A list of ModelBinderDescriptors.</param>
+        public static void RemoveTypesOf<TInstance>(
+            [NotNull] this IList<ModelBinderDescriptor> descriptors)
+            where TInstance : class, IModelBinder
+        {
+            for (int i = descriptors.Count - 1; i >= 0; i--)
+            {
+                if (descriptors[i].OptionType == typeof(TInstance))
```

Comment:
if anyone sub-classes `BinderTypeBasedModelBinder` and calls `RemoveTypesOf<BinderTypeBasedModelBinder>`, the sub-classed instances will not be removed because those `OptionDescriptor<TOption>`s will have `OptionType=={that subclass}` even if created with an explicit `TOption`.  `OptionType` is very concerned about the instance's type when `Instance` is non-`null`.

probably not an issue as long as we don't expect / want anyone to create complicated `IModelBinder` (really `TOption` in general) type hierarchies. but worth `<remarks/>` in the XML comments about `exactly <typeparamref name="TInstance"/>`.